demos/security-trifecta/src/core/PrivateRoute.tsx

```diff
@@ -1,11 +1,11 @@
 import { ReactNode } from 'react';
-import { Redirect, Route, useLocation } from 'react-router';
-import { useAuth } from './AuthProvider';
+import { Redirect } from 'react-router';
+import { getSnapshot } from '../utils/session-vault';
 
 type Props = { children?: ReactNode };
 
 export const PrivateRoute = ({ children }: Props) => {
-  const { session } = useAuth();
+  const session = getSnapshot();
```

Comment:
Does this break if you use `useAuth()` instead of the snapshot directly?

Reply:
I was getting inconsistent behavior after logging in if I do `useAuth` instead of the `getSnapshot` so I'm leaving that as is until I can dig more into it

Reply:
To be specific, sometimes it would have me create the passcode then go back to the login page and have to start over